requirements.txt

```diff
@@ -2,6 +2,7 @@
 # Make sure you also install the non-Python dependencies described in
 # https://github.com/google/pytype/blob/master/CONTRIBUTING.md#pytype-dependencies.
 attrs>=21.4.0
+cmake>=3.16
```

Comment:
I think we shouldn't be adding cmake here.

1. As I understand, it's a build-only dependency, whereas this file lists run-time dependencies. It is not necessary to install cmake to use pytype (I *think*)

2. It is rather controversial to install cmake from PyPI, or at least I haven't heard of this before.

3. For GitHub Actions it shouldn't be a problem, cmake is always available there

Reply:
I agree that `cmake` should at least be moved to `build-system.requires` in `pyproject.toml`, which already includes tools such as `ninja`.

I'd consider `cmake` and `ninja` to be equally controversial, being build tools external to Python's ecosystem, and `ninja` is already being installed from PyPI here.

Reply:
I see, indeed they are about equal. Sure.

Oh and also this file *is* for dev dependencies, we are not adding it to install dependencies.

OK, this is fine.

Reply:
My mistake. If it's only a dev dependency than it should be mentioned in `CONTRIBUTING.md` instead, which CMake already is. I'd agree with removing it from the requirements.txt in this case.

The CMake version in `CONTRIBUTING.md` should be bumped to `3.16` since that's the current minimum required version.

Reply:
Luckily, I already bumped the CMake version in CONTRIBUTING.md to 3.16! :)

I don't 100% understand which of these dependencies are used when, but moving cmake to build-system.requires works (moving ninja out of requirements.txt in the same way doesn't also work), so I have done so. Uh, by "works" I mean the CI fails on the same further step, of course.

Reply:
`requirements.txt`: suggested dependencies for package developers, in this case the ones working on pytype directly, would be installed into your local venv of your cloned project, sometimes automatically installed by your IDE.

`build-system.requires`: dependencies required to run `setup.py` or whichever `build-system.build-backend` is specified (currently setup.py), but are not needed to import the package normally. Do not move a dependency into `build-system.requires` unless the Python build step fails because that dependency is missing.

I haven't touched this project since #1154, having multiple suggested ways to build the project is a little confusing. `pip install -e .` would use `build-system.requires`, but the non-standard `python build_scripts/build.py` would expect the dependencies from `requirements.txt` instead. You are going to get different results depending on which build method is being used. I'm too much of an outsider to give any suggestions or advice on this, and my earlier advice may have been premature.

Reply:
I'm sorry, I think I'm not ready to accept these changes to build requirements within this pull request.

The decision how exactly cmake gets pulled in does not directly affect Windows support, and the situation is not unique to Windows; the support can be achieved without changing this. But it *does* affect every system in ways that we haven't fully understood yet.

So in the interest of getting this merged more easily, I am going to revert those changes from this pull request.

Reply:
Sounds good. Thanks!